test/components/link_test.exs

```diff
@@ -180,24 +180,79 @@ defmodule Surface.Components.LinkTest do
                ~s[<a data-method="put" data-to="/world" rel="nofollow" href="/world">hello</a>]
     end
 
+    test "link with :do contents" do
+      html =
+        render_surface do
+          ~H"""
+          <Link to="/hello">
+            {{ Phoenix.HTML.Tag.content_tag(:p, "world") }}
+          </Link>
+          """
+        end
+
+      assert html == """
+             <a href="/hello">
+               <p>world</p>
+             </a>
+             """
+    end
+
     test "link with scheme" do
-      html = render_surface(do: ~H[<Link to="/javascript:alert(<1>)" />])
-      assert html =~ ~s[<a href="/javascript:alert(&lt;1&gt;)"></a>]
+      html = render_surface(do: ~H[<Link label="foo" to="/javascript:alert(<1>)" />])
+      assert html =~ ~s[<a href="/javascript:alert(&lt;1&gt;)">foo</a>]
 
-      html = render_surface(do: ~H[<Link to={{ {:safe, "/javascript:alert(<1>)"} }} />])
-      assert html =~ ~s[<a href="/javascript:alert(<1>)"></a>]
+      html =
+        render_surface(do: ~H[<Link label="foo" to={{ {:safe, "/javascript:alert(<1>)"} }} />])
+
+      assert html =~ ~s[<a href="/javascript:alert(<1>)">foo</a>]
 
-      html = render_surface(do: ~H[<Link to={{ {:javascript, "alert(<1>)"} }} />])
-      assert html =~ ~s[<a href="javascript:alert(&lt;1&gt;)"></a>]
+      html = render_surface(do: ~H[<Link label="foo" to={{ {:javascript, "alert(<1>)"} }} />])
+      assert html =~ ~s[<a href="javascript:alert(&lt;1&gt;)">foo</a>]
 
-      html = render_surface(do: ~H[<Link to={{ {:javascript, 'alert(<1>)'} }} />])
-      assert html =~ ~s[<a href="javascript:alert(&lt;1&gt;)"></a>]
+      html = render_surface(do: ~H[<Link label="foo" to={{ {:javascript, 'alert(<1>)'} }} />])
+      assert html =~ ~s[<a href="javascript:alert(&lt;1&gt;)">foo</a>]
+
+      html =
+        render_surface(do: ~H[<Link label="foo" to={{ {:javascript, {:safe, "alert(<1>)"}} }} />])
 
-      html = render_surface(do: ~H[<Link to={{ {:javascript, {:safe, "alert(<1>)"}} }} />])
-      assert html =~ ~s[<a href="javascript:alert(<1>)"></a>]
+      assert html =~ ~s[<a href="javascript:alert(<1>)">foo</a>]
 
-      html = render_surface(do: ~H[<Link to={{ {:javascript, {:safe, 'alert(<1>)'}} }} />])
-      assert html =~ ~s[<a href="javascript:alert(<1>)"></a>]
+      html =
+        render_surface(do: ~H[<Link label="foo" to={{ {:javascript, {:safe, 'alert(<1>)'}} }} />])
+
+      assert html =~ ~s[<a href="javascript:alert(<1>)">foo</a>]
+    end
+
+    test "link with invalid args" do
+      msg = "expected non-nil value for :to in <Link />"
+
+      assert_raise ArgumentError, msg, fn ->
+        render_surface(do: ~H[<Link label="foo" opts={{ bar: "baz" }} />])
+      end
+
+      # msg = "link/2 requires a keyword list as second argument"
```

Comment:
@msaraiva this Phoenix test is very specific to their implementation, not sure it makes sense to force it here, what do you think?

Reply:
Just remove it.